File: deploy_new.js
const algosdk = require('algosdk');
const fs = require('fs');

async function main() {
    const mnemonic = "bracket inform cricket pact impact mask misery quantum dismiss giant fog mechanic category royal east actual gold easily snow laundry vast gown crater absent denial";
    const account = algosdk.mnemonicToSecretKey(mnemonic);
    const adminAddr = account.addr.toString();
    console.log('Admin:', adminAddr);

    const approval = fs.readFileSync('projects/smart_contracts/dojo_registry/artifacts/DojoRegistry.approval.teal');
    const clear = fs.readFileSync('projects/smart_contracts/dojo_registry/artifacts/DojoRegistry.clear.teal');
    console.log('Approval size:', approval.length, 'bytes');

    const client = new algosdk.Algodv2('', 'https://testnet-api.algonode.cloud', '443');
    const accountInfo = await client.accountInformation(adminAddr).do();
    console.log('Balance:', Number(accountInfo.amount) / 1e6, 'ALGO');

    const params = await client.getTransactionParams().do();
    params.fee = 70000;
    params.flatFee = true;
    params.lastValid = Number(params.firstValid) + 2000;

    // Try BARE app create (no app args) - then call create method separately
    const txn = algosdk.makeApplicationCreateTxnFromObject({
        sender: adminAddr,
        suggestedParams: params,
        onComplete: algosdk.OnApplicationComplete.NoOpOC,
        approvalProgram: approval,
        clearProgram: clear,
        numGlobalInts: 1,
        numGlobalByteSlices: 1,
        numLocalInts: 0,
        numLocalByteSlices: 0,
        extraPages: 4,
        appArgs: [],  // No args - bare app create
    });

    console.log('Signing transaction...');
    const signedTxn = txn.signTxn(account.sk);
    
    console.log('Sending transaction...');
    const sendResult = await client.sendRawTransaction(signedTxn).do();
    console.log('TxID:', sendResult.txid);

    console.log('Waiting for confirmation...');
    const confirmed = await algosdk.waitForConfirmation(client, sendResult.txid, 15);
    console.log('App ID:', confirmed.applicationIndex);
    
    // Now call create() method to set admin
    console.log('\nCalling create() to set admin...');
    const appId = confirmed.applicationIndex;
    
    const callParams = await client.getTransactionParams().do();
    callParams.fee = 2000;
    callParams.flatFee = true;
    
    const callTxn = algosdk.makeApplicationCallTxnFromObject({
        sender: adminAddr,
        suggestedParams: callParams,
        appId: appId,
        onComplete: algosdk.OnApplicationComplete.NoOpOC,
        appArgs: [Buffer.from('create'), Buffer.from(adminAddr, 'utf8')],
    });
    
    const signedCall = callTxn.signTxn(account.sk);
    const callResult = await client.sendRawTransaction(signedCall).do();
    console.log('Call TxID:', callResult.txid);
    
    await algosdk.waitForConfirmation(client, callResult.txid, 10);
    
    console.log('\n=== DEPLOYMENT SUCCESSFUL ===');
    console.log('DojoRegistry App ID:', appId);
    console.log('\nUpdate .env: DOJO_REGISTRY_APP_ID=' + appId);
}

main().catch(console.error);

File: deploy_new.py
#!/usr/bin/env python3
import os
import sys
from dotenv import load_dotenv

load_dotenv()

import algosdk
from algosdk.v2client import algod
from algosdk.transaction import ApplicationCreateTransaction, OnComplete
from algosdk import wait_for_confirmation

# Load mnemonic from .env
mnemonic = os.getenv("ADMIN_PASSPHRASE")
account = algosdk.mnemonic_to_secret_key(mnemonic)
admin_address = account.addr
print(f"Admin: {admin_address}")

# Load contracts
approval = open(
    "projects/smart_contracts/dojo_registry/artifacts/DojoRegistry.approval.teal"
).read()
clear = open(
    "projects/smart_contracts/dojo_registry/artifacts/DojoRegistry.clear.teal"
).read()
print(f"Approval TEAL size: {len(approval)} bytes")

# Create client
algod_token = os.getenv("ALGOD_TOKEN", "")
algod_server = os.getenv("ALGOD_SERVER", "https://testnet-api.algonode.cloud")
algod_port = os.getenv("ALGOD_PORT", "443")

client = algod.AlgodClient(algod_token, f"{algod_server}:{algod_port}")

# Get account info
acct_info = client.account_info(admin_address)
balance = acct_info.get("amount", 0) / 1e6
print(f"Balance: {balance:.4f} ALGO")

# Get suggested params
params = client.suggested_params()
params.fee = 20000  # Higher fee for large contract
params.flat_fee = True

# TEAL is text, encode to bytes
app_args = [admin_address.encode()]

txn = ApplicationCreateTransaction(
    sender=admin_address,
    sp=params,
    on_complete=OnComplete.NoOpOC,
    approval_program=approval.encode(),
    clear_program=clear.encode(),
    global_schema={"num_byte_slices": 1, "num_uints": 1},
    local_schema={"num_byte_slices": 0, "num_uints": 0},
    extra_pages=1,  # Allow larger contract
    app_args=app_args,
)

# Sign
signed = txn.sign(account.private_key)

# Send
tx_id = client.send_transaction(signed)
print(f"Transaction ID: {tx_id}")

# Wait for confirmation
confirmed = wait_for_confirmation(client, tx_id, timeout=10)
app_id = confirmed.get("application-index")
print(f"\n=== SUCCESS ===")
print(f"New App ID: {app_id}")
print(f"\nUpdate .env with: DOJO_REGISTRY_APP_ID={app_id}")
